tools/data-api/internal/repositories/generator_models.go

```diff
@@ -14,3 +14,64 @@ type Value struct {
 	Value       string  `json:"Value"`
 	Description *string `json:"Description,omitempty"`
 }
+
+type Model struct {
+	Fields          []Field `json:"Fields"`
+	IsDiscriminator *bool   `json:"IsDiscriminator,omitempty"`
+	ParentModelName *string `json:"ParentModelName,omitempty"`
+	ValueForType    *string `json:"ValueForType,omitempty"`
+}
+
+type Field struct {
+	Name             string           `json:"Name"`
+	JsonName         string           `json:"JsonName"`
+	Required         *bool            `json:"Required,omitempty"`
+	Optional         *bool            `json:"Optional,omitempty"`
+	ObjectDefinition ObjectDefinition `json:"ObjectDefinition"`
+	MinItems         *int             `json:"MinItems,omitempty"`
+	MaxItems         *int             `json:"MaxItems,omitempty"`
+	DateFormat       *string          `json:"DateFormat,omitempty"`
+	ProvidesTypeHint *bool            `json:"ProvidesTypeHint,omitempty"`
+}
+
+type Operation struct {
+	Name                             string           `json:"Name"`
+	ContentType                      string           `json:"ContentType,omitempty"`
+	ExpectedStatusCodes              []int            `json:"ExpectedStatusCodes,omitempty"`
+	FieldContainingPaginationDetails string           `json:"FieldContainingPaginationDetails,omitempty"`
+	LongRunning                      bool             `json:"LongRunning,omitempty"`
+	HTTPMethod                       string           `json:"HTTPMethod,omitempty"`
+	OptionsObject                    OptionsObject    `json:"OptionsObject,omitempty"`
+	Options                          []Option         `json:"Options,omitempty"`
+	ResourceId                       string           `json:"ResourceId,omitempty"`
+	RequestObject                    ObjectDefinition `json:"RequestObject,omitempty"`
+	ResponseObject                   ObjectDefinition `json:"ResponseObject,omitempty"`
```

Comment:
these are both Optional therefore need to be defined as pointers:

```suggestion
	RequestObject                    *ObjectDefinition `json:"RequestObject,omitempty"`
	ResponseObject                   *ObjectDefinition `json:"ResponseObject,omitempty"`
```